### PR TITLE
Improve build stability by ensuring we aren't double publishing everywhere

### PR DESCRIPTION
**Pull request by marcpopMSFT**

### CODE REVIEW

**Comment by mmitche** [comment]
I think this looks good. Do you have an official build test that I can peek at?

**Comment by mmitche** [comment]
Double publish is the new double free

**Comment by marcpopMSFT** [comment]
@mmitche here's the CI build I ran before cherry-picking over: https://dev.azure.com/dnceng/internal/_build/results?buildId=1993258&view=results

**Comment by mmitche** [comment]
Official looks good.

**Comment by marcpopMSFT** [comment]
Had to fix the richnav one but this should be good now.

File: .vsts-ci.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ variables:
     value: Roslyn-Project-System
   - name: _DotNetPublishToBlobFeed
     value: false
+  - name: CIBUILD
```
Comment (mmitche) [inline .vsts-ci.yml:21]: Nit: _CIBuild to align with other parameters.

**Comment by marcpopMSFT** [comment]
@mmitche got it passing post PR feedback, ready for review.

**Review by mmitche — APPROVED** [review]

**Pull request merged**